lib/jit_preloader/active_record/base.rb

```diff
@@ -131,11 +132,20 @@ def has_many_aggregate(assoc, name, aggregate, field, table_alias_name: nil, def
             end
             group_by = "#{table_reference}.#{aggregate_association.foreign_key}"
 
-            preloaded_data = Hash[association_scope
-              .where(conditions)
-              .group(group_by)
-              .send(aggregate, field)
-            ]
+            if JitPreloader.query_dive_limit
+              slices = primary_ids.each_slice(JitPreloader.query_dive_limit)
+            end
+
+            preloaded_data = {}
+            slices.each do |slice|
+              conditions[table_reference][aggregate_association.foreign_key] = slice
```

Comment:
Would it be clearer if we didn't modify `conditions` variable?  Like, I'm suggesting 
```
              data = Hash[association_scope
                            .where(conditions[table_reference].merge(aggregate_association.foreign_key => slice))
                            .group(group_by)
                            .send(aggregate, field)
              ]
```

That way, if future writers modify this, we don't have a conditions object that looks like it works, but might not in some cases.

Reply:
That's a good idea! let me dig into that.